Automattic/newspack-popups
# feat: segmentation match logic

### All Submissions:






### Changes proposed in this Pull Request:



Implements a JS-based segmentation API using the criteria system. Removes AMP markup and config JSON. Replaces AMP polyfill JS with vanilla DOM-based JS.

Not working right now (will be resolved in future PRs):

- Previewing single prompts or segments from the Campaigns dashboard (interestingly, previewing a single prompt from the editor works)
- Donor landing page setting
- Mailchimp donor merge fields setting
- Non-interactive mode setting (this may be removed since the segmentation API is no longer server-based)

Closes `1204189396255901/1204261432690236`.

### How to test the changes in this Pull Request:

1. Create a variety of segments and both overlay/inline prompts targeting different criteria. Ensure the prompts have a variety of settings, especially when it comes to frequency, and trigger (for overlays) settings.
2. View on the front-end and confirm that all prompts display as expected according to frequency settings, scroll/delay triggers, and segment criteria.
3. Also test Above Header, Custom Placement, Manual Only placements, and shortcodes in post/page content and in widgets.

### Other information:
